[DOCS] AUTH-509 Reducing permissions for unauthenticated users for apiserver access (#75173)
openshift/openshift-docs: modules/unauthenticated-users-cluster-role-binding.adoc

// Module included in the following assemblies:
//
// * authentication/impersonating-system-admin.adoc
// * cicd/builds/triggering-builds-build-hooks.adoc
// * authentication/tokens-scoping.adoc
// * authentication/managing-oauth-access-tokens.adoc
// * post_installation_configuration/preparing-for-users.adoc

:_mod-docs-content-type: PROCEDURE
[id="unauthenticated-users-cluster-role-bindings_{context}"]
= Adding unauthenticated groups to `ClusterRoleBindings`

> As a cluster administrator, you can add unauthenticated users to the following ClusterRoleBindings in {product-title}.

Reviewer: Same about using "cluster role bindings" throughout - I won't point out any more instances in this PR

> . `system:scope-impersonation`

Reviewer: Same about unnumbered bullet

. `system:webhook`
. `system:oauth-token-deleter`
. `self-access-reviewer`

[IMPORTANT]
====
Always verify compliance with your organization's security standards when modifying role bindings, particularly when including unauthenticated access.
====

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` role.

> * The OpenShift CLI (oc) is installed.

Reviewer: `oc` in backticks. Or you can also use the {oc-first} attribute, which will already have the proper way to write it.

Also recommend writing so that the prereq items are more parallel. Could do:

```suggestion
* You have installed the OpenShift CLI (`oc`).
```

.Procedure

> .Add unauthenticated groups to a `ClusterRoleBinding`

Reviewer: Per the procedure template, there shouldn't be any other additional headings under the `.Procedure` heading

```suggestion

```

. Create a YAML file named `add-<cluster-role-binding>-unauth.yaml` and add the following content:
+
[source,yaml]
----
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
 annotations:
   rbac.authorization.kubernetes.io/autoupdate: "true"

>  name: restore-<cluster-role-binding>-unauth

Reviewer: For both of these places that call out to replace this with the cluster role binding, I'd add a callout or something to say to make sure to replace them with the one you want to update.

roleRef:
 apiGroup: rbac.authorization.k8s.io
 kind: ClusterRole

>  name: <cluster-role-binding>

Reviewer: All of these replaceable values should use underscores

```suggestion
 name: <cluster_role_binding>
```

subjects:
 - apiGroup: rbac.authorization.k8s.io
   kind: Group
   name: system:unauthenticated
----
. Apply the configuration using the CLI:
+

> [source,bash]

Reviewer: All terminal commands should use `source,terminal`

```suggestion
[source,terminal]
```

----

> oc apply -f add-<cluster-role-binding>.yaml

Reviewer: Missing the prompt

```suggestion
$ oc apply -f add-<cluster_role_binding>.yaml
```

// Module included in the following assemblies:
//
// * authentication/impersonating-system-admin.adoc
// * cicd/builds/triggering-builds-build-hooks.adoc
// * authentication/tokens-scoping.adoc
// * authentication/managing-oauth-access-tokens.adoc
// * post_installation_configuration/preparing-for-users.adoc

:_mod-docs-content-type: PROCEDURE
[id="unauthenticated-users-cluster-role-bindings_{context}"]
= Adding unauthenticated groups to cluster role bindings

> As a cluster administrator, you can add unauthenticated users to the following cluster role bindings in {product-title}. Unauthenticated users do not have access to most cluster role bindings by default.

Reviewer 2: ```suggestion
As a cluster administrator, you can add unauthenticated users to the following cluster roles in {product-title} by creating a (cluster-)role-binding. Unauthenticated users do not have access to non-public cluster roles.
```

Author: Can I say "private cluster roles" instead of "non-public?" Or is there a difference between the two that is relevant here?

Reviewer 2: I think the emphasis is on "public" as it might come as a surprise to everyone that there are actually 2 public cluster roles. They have "public" in their names as well... If we only say "private cluster roles", the immediate question that arises is: "are not all of them private"?

* `system:scope-impersonation`
* `system:webhook`
* `system:oauth-token-deleter`

> * `self-access-reviewer`

Reviewer 2: We should create individual entries for all 4 cases. A combination of those 4 cases could be an entry that helps administrators to have the exact same setup as before, but for, e.g. in the case of [Build triggers](https://docs.openshift.com/container-platform/4.15/cicd/builds/triggering-builds-build-hooks.html#builds-webhook-triggers_triggering-builds-build-hooks), the only cluster role of interest is `system:webhook`. Showing all 4 of them adds a lot of confusion.

[IMPORTANT]
====
Always verify compliance with your organization's security standards when modifying unauthenticated access.
====

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` role.
* You have installed the OpenShift CLI (`oc`).

.Procedure

[Note]
====

> Replace the `<cluster_role_binding>` variable with the cluster role binding you want to update.

Reviewer 2: ```suggestion
Replace the `<cluster_role>` variable with the cluster role binding you want to add.
```

====

. Create a YAML file named `add-<cluster_role_binding>-unauth.yaml` and add the following content:
+
[source,yaml]
----
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
 annotations:
   rbac.authorization.kubernetes.io/autoupdate: "true"

>  name: <cluster_role>-unauth

Reviewer 2: We could streamline the naming here:

```suggestion
 name: <cluster_role>access-unauthenticated
```

>  name: restore-<cluster_role_binding>-unauth

Reviewer 2: We could drop the `restore` part. I will ask if there is any dedicated naming convention for cluster-role-bindings.
What you mean in this case is not a `cluster-role-binding`, but a `cluster-role`.

roleRef:
 apiGroup: rbac.authorization.k8s.io
 kind: ClusterRole

>  name: <cluster_role_binding>

Reviewer 2: This is the name of the Cluster Role, not the binding. So it would need to be 

```yaml
  name: "system:webhook"
```.

It is important to put `system:webhook` in `"`, otherwise it is misinterpreted.

Reviewer 2: ```mermaid
graph LR
    A[system:unauthenticated]
    C[system:webhook]

    A -- restore-webhook-unauth --> C
```

//
// * authentication/impersonating-system-admin.adoc
// * authentication/tokens-scoping.adoc
// * authentication/managing-oauth-access-tokens.adoc
// * post_installation_configuration/preparing-for-users.adoc

:_mod-docs-content-type: PROCEDURE
[id="unauthenticated-users-cluster-role-bindings_{context}"]
= Adding unauthenticated groups to cluster roles

As a cluster administrator, you can add unauthenticated users to the following cluster roles in {product-title} by creating a cluster role binding. Unauthenticated users do not have access to non-public cluster roles. This should only be done in specific use cases when necessary.

You can add unauthenticated users to the following cluster roles:

* `system:scope-impersonation`
* `system:webhook`
* `system:oauth-token-deleter`
* `self-access-reviewer`

[IMPORTANT]
====
Always verify compliance with your organization's security standards when modifying unauthenticated access.
====

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` role.
* You have installed the OpenShift CLI (`oc`).

.Procedure

[Note]
====
Replace the `<cluster_role>` variable with the cluster role you want to update.

> ====

Reviewer 3: This section isn't rendering correctly. Maybe because _Note_ isn't all caps? Or, it doesn't like the admonition after the heading?
![image](https://github.com/openshift/openshift-docs/assets/25209155/0cd40ca8-f3b2-4ae3-8af4-69e5f3a95ce0)

. Create a YAML file named `add-<cluster_role>-unauth.yaml` and add the following content:
+
[source,yaml]
----
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
 annotations:
   rbac.authorization.kubernetes.io/autoupdate: "true"
 name: <cluster_role>access-unauthenticated
roleRef:
 apiGroup: rbac.authorization.k8s.io
 kind: ClusterRole
 name: <cluster_role>
subjects:
 - apiGroup: rbac.authorization.k8s.io
   kind: Group
   name: system:unauthenticated

> ----

Reviewer 4: The Note is rendering a little... weird:
![Screenshot 2024-06-21 at 1 07 19 PM](https://github.com/openshift/openshift-docs/assets/112414607/ec538143-cc6f-425b-ad24-9d662ea0e91b)

At this point, I would consider three options:

1. Moving the note underneath the yaml
2. Removing the note entirely, I'm not sure but I think the meaning of the input variable is already strongly implied.
3. Using callout labels in the yaml and hope that the user knows to apply the same info when creating the file name for the yaml file.

Reviewer 4: You could also play around with other options like adding a + in between .Procedure and the note, but I understand there's not much time right now

Author: I think you're right about it being heavily implied and it's probably overkill.

> . Apply the configuration using the CLI:

Reviewer 4: Per the [docs guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command) we need to explicitly mention whenever a command is run:
```suggestion
. Apply the configuration by running the following command:
```
